Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1085 @@
+                                         || flag_info_face_z(i + i_n, j + j_n, k) == 2)
+                                    && flag_ext_face_z(i, j, k)) {
+                                    Sz_mod(i + i_n, j + j_n, k) -= Sz_ext;
+                                    // Insert the index of the face info
+                                    *(borrowing_z_inds + ps) = ps;
+                                    // Store the information about the intruded face in the dataset of the
+                                    // faces which are borrowing area
+                                    FaceInfoBox::addConnectedNeighbor(i_n, j_n, ps,
+                                                                      borrowing_z_neigh_faces);
+                                    *(borrowing_z_area+ps) = Sz_ext;
+
+                                    flag_info_face_z(i + i_n, j + j_n, k) = 2;
+                                    // Add the area to the intruding face.
+                                    Sz_mod(i, j, k) = Sz(i, j, k) + Sz_ext;
+                                    flag_ext_face_z(i, j, k) = false;
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+        amrex::Scan::Type::exclusive);
+
+
+    }
+
+    return {nelems_x, nelems_y, nelems_z};
+#else
+    return {0, 0, 0};
+#endif
+}
+
+
+void
+WarpX::ComputeEightWaysExtensions(amrex::Array1D<int, 0, 2> temp_inds) {
+#ifdef AMREX_USE_EB
+    const int n_borrow_offset_x = temp_inds(0);
+    const int n_borrow_offset_y = temp_inds(1);
+    const int n_borrow_offset_z = temp_inds(2);
+
+    auto const &cell_size = CellSize(maxLevel());
+    int idim = 0;
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+
+        amrex::Box const &box = mfi.validbox();
+
+        auto const &Sx = m_face_areas[maxLevel()][idim]->array(mfi);
+        auto const &flag_ext_face_x = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+        auto const &flag_info_face_x = m_flag_info_face[maxLevel()][idim]->array(mfi);
+        auto &borrowing_x = (*m_borrowing[maxLevel()][idim])[mfi];
+        auto const &borrowing_x_inds_pointer = borrowing_x.inds_pointer.array();
+        auto const &borrowing_x_size = borrowing_x.size.array();
+        amrex::Long ncells = box.numPts();
+        int* borrowing_x_inds = borrowing_x.inds.data();
+        uint8_t* borrowing_x_neigh_faces = borrowing_x.neigh_faces.data();
+        amrex::Real* borrowing_x_area = borrowing_x.area.data();
+
+        auto const &Sx_mod = m_area_mod[maxLevel()][idim]->array(mfi);
+        const auto &ly = m_edge_lengths[maxLevel()][1]->array(mfi);
+        const auto &lz = m_edge_lengths[maxLevel()][2]->array(mfi);
+        const amrex::Real dy = cell_size[1];
+        const amrex::Real dz = cell_size[2];
+        int nelems_x;
+
+        nelems_x = amrex::Scan::PrefixSum<int>(ncells,
```

Comment:
Since `nelems_x` is unused, maybe this could become
```suggestion
        amrex::Scan::PrefixSum<int>(ncells,
```

Reply:
Same thing with `nelems_y`, `nelems_z` in the same function.